server/controller.py

```diff
@@ -842,13 +842,26 @@ def mempool_get_fee_histogram(self):
         to the daemon's memory pool.'''
         return await self.daemon_request('relayfee')
 
-    async def transaction_get(self, tx_hash):
+    async def transaction_get(self, tx_hash, verbose=False):
         '''Return the serialized raw transaction given its hash
 
         tx_hash: the transaction hash as a hexadecimal string
+        verbose: (optional, default=False), enable\disable verbose mode
         '''
         self.assert_tx_hash(tx_hash)
-        return await self.daemon_request('getrawtransaction', tx_hash)
+        transaction = await self.daemon_request('getrawtransaction', tx_hash, int(verbose))
+        if not transaction:
+            return
+        if verbose and isinstance(transaction, dict):
+            try:
+                return {
+                    'hex': transaction['hex'],
+                    'blockhash': transaction.get('blockhash')
+                }
+            except KeyError:
+                self.logger.exception('verbose mode failed')
+                raise RPCError("error in verbose mode")
+        return transaction
```

Comment:
As I indicated yesterday let's simply return what the network daemon returns and keep it simple.  Just change line 851 to 

return await self.daemon_request('getrawtransaction', tx_hash, verbose)

so it's only a 2-line change - that line and the async def line.

The Daemon class infrastructure will automatically return a good error if an error occurs; I designed things so being a user of the class is simple.

You will probably need to update your tests.

Reply:
Looks like tests are ok, since the response is mocked and the behaviour is almost the same.